### PR TITLE
feat(git): Add helpers for platform-native commit

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- `pushCommitAsRef` allows to create remote non-branch refs
- `listCommitTree` obtains tree contents from the local commit
- Extract `newlineRegex` and refactor usages of `split()` function
- Rename `prevCommitSha` to `parentCommitSha`

## Context:

- Ref: #11537

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -893,3 +898,30 @@ export function getUrl({
...
+  /^(?<mode>\d{6})\s+(?<type>blob|tree)\s+(?<sha>[0-9a-f]{40})\s+(?<path>.*)$/
+);
+
+export async function listCommitTree(commitSha: string): Promise<TreeItem[]> {
+  const treeSha = (await git.raw(['cat-file', '-p', commitSha])).slice(5, 45);
+  const contents = await git.raw(['cat-file', '-p', treeSha]);
+  const lines = contents.split(newlineRegex);
+  const result: TreeItem[] = [];
+  for (const line of lines) {
+    const matchGroups = line.match(treeItemRegex)?.groups;
```
Comment (viceice) [inline lib/util/git/index.ts:920]: Use Regex.exec instead

File: lib/util/git/index.ts [hunk 2]
```diff
@@ -893,3 +896,30 @@ export function getUrl({
...
+
+export async function pushCommitAsRef(
+  commitSha: string,
+  refName: string
+): Promise<void> {
+  await git.raw(['update-ref', refName, commitSha]);
+  await git.raw(['push', '--force', 'origin', refName]);
+}
+
+const treeItemRegex = regEx(
```
Comment (viceice) [inline lib/util/git/index.ts:908]: should we create a `./util.ts` and move there, as this file is already pretty big?

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

File: lib/util/git/index.ts (hunk 2 above)
Comment (zharinov) [reply lib/util/git/index.ts:908]: Then we should do something with internal `simple-git` instance in `index.ts`

Comment (rarkins) [reply lib/util/git/index.ts:908]: Let's not delay further if this is a major refactor

File: lib/util/git/index.ts [hunk 3]
```diff
@@ -893,3 +896,30 @@ export function getUrl({
...
+  /^(?<mode>\d{6})\s+(?<type>blob|tree)\s+(?<sha>[0-9a-f]{40})\s+(?<path>.*)$/
+);
+
+export async function listCommitTree(commitSha: string): Promise<TreeItem[]> {
+  const treeSha = (await git.raw(['cat-file', '-p', commitSha])).slice(5, 45);
+  const contents = await git.raw(['cat-file', '-p', treeSha]);
```
Comment (rarkins) [inline lib/util/git/index.ts:914]: Document this and any assumptions it relies on

Comment (rarkins) [reply lib/util/git/index.ts:914]: Can maybe be more sustainable to split the contents, filter for `tree x` and then extract `x`.

File: lib/util/git/index.ts (hunk 2 above)
Comment (viceice) [reply lib/util/git/index.ts:908]: I only meant the Regex constant, not the functions. 🤗

**Comment by viceice** [comment]
Needs deconfliction

**Review by rarkins — APPROVED** [review]

**Pull request merged**